server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -114,13 +115,15 @@ private long calculateQuantity(long quantity, Product product, String upstreamPo
      */
     public List<Pool> createAndEnrichPools(Pool masterPool, List<Pool> existingPools) {
         List<Pool> pools = new LinkedList<Pool>();
-        masterPool.setQuantity(calculateQuantity(masterPool.getQuantity(),
-            masterPool.getProduct(), masterPool.getUpstreamPoolId()));
 
-        String virtOnly = masterPool.getProductAttributeValue(Product.Attributes.VIRT_ONLY);
+        masterPool.setQuantity(calculateQuantity(masterPool.getQuantity(), masterPool.getProduct(),
+            masterPool.getUpstreamPoolId()));
+
         // The following will make virt_only a pool attribute. That makes the
         // pool explicitly virt_only to subscription manager and any other
         // downstream consumer.
+        // TODO: Should we remove VIRT_ONLY from the pool if the attribute isn't present on the product?
```

Comment:
I'd like to resolve this question before we merge.  To me the answer seems like "yes" but I'd be interested to hear @vritant opinion.

Reply:
yes, looked at the code, cant find a reason as to why that would be a problem.

Reply:
Alright. Comment removed and replaced with a line that removes the pool attribute if the product attribute lookup fails.